qtpy/_patch/qfiledialog.py

```diff
@@ -0,0 +1,159 @@
+                         selectedfilter='', options=None):
+        """
+        Wrapper around QtGui.QFileDialog.getOpenFileNames static method.
+
+        Returns a tuple (filenames, selectedfilter) -- when dialog box is
+        canceled, returns a tuple (empty list, empty string).
+
+        Compatible with PyQt >=v4.4 (API #1 and #2) and PySide >=v1.0.
+        """
+        return _qfiledialog_wrapper('getOpenFileNames',
+                                    parent=parent,
+                                    caption=caption,
+                                    basedir=basedir,
+                                    filters=filters,
+                                    selectedfilter=selectedfilter,
+                                    options=options)
+
+
+    def _getSaveFileName(parent=None, caption='', basedir='', filters='',
+                        selectedfilter='', options=None):
+        """
+        Wrapper around QtGui.QFileDialog.getSaveFileName static method.
+
+        Returns a tuple (filename, selectedfilter) -- when dialog box is
+        canceled, returns a tuple of empty strings.
+
+        Compatible with PyQt >=v4.4 (API #1 and #2) and PySide >=v1.0.
+        """
+        return _qfiledialog_wrapper('getSaveFileName',
+                                    parent=parent,
+                                    caption=caption,
+                                    basedir=basedir,
+                                    filters=filters,
+                                    selectedfilter=selectedfilter,
+                                    options=options)
+
+
+    def _getExistingDirectory(parent=None, caption='', basedir='',
+                             options=QFileDialog.ShowDirsOnly):
+        """
+        Wrapper around QtGui.QFileDialog.getExistingDirectory static method.
+
+        Compatible with PyQt >=v4.4 (API #1 and #2) and PySide >=v1.0.
+        """
+        # Calling QFileDialog static method
+        if sys.platform == "win32":
+            # On Windows platforms: redirect standard outputs
+            _temp1, _temp2 = sys.stdout, sys.stderr
+            sys.stdout, sys.stderr = None, None
+        try:
+            result = QFileDialog.getExistingDirectory(parent,
+                                                      caption,
+                                                      basedir,
+                                                      options)
+        finally:
+            if sys.platform == "win32":
+                # On Windows platforms: restore standard outputs
+                sys.stdout, sys.stderr = _temp1, _temp2
+        if not is_text_string(result):
+            # PyQt API #1
+            result = to_text_string(result)
+        return result
+
+    QFileDialog.getOpenFileName2 = _getOpenFileName
+    QFileDialog.getOpenFileNames2 = _getOpenFileNames
+    QFileDialog.getSaveFileName2 = _getSaveFileName
+    QFileDialog.getExistingDirectory2 = _getExistingDirectory
```

Comment:
No they don't, but if we replace the normal methods, the return is changed, so it would be weird.

We can warn QFileDialog.getExistingDirectory to better use QFileDialog.getExistingDirectory2. 

Unfortunately I cant run any test on this, cause I am either stuch in the exec loop or segfaulting. I think that in general is a very bad idea/bad practice to use this methods that call exec/modal directly. It is better and safer (as seen by the latest fixes of @rlaverde in spyder) to avoid using these methods :-\